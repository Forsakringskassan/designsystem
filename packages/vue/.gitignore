# vite dev
/src/vite-dev/app.vue
