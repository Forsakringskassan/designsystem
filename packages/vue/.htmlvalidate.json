{
    "extends": ["@fkui/vue:recommended"],
    "plugins": ["./htmlvalidate"],
    "transform": {
        "^.*\\.spec.ts$": "html-validate-vue:html"
    }
}
