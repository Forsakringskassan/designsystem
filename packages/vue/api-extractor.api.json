{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "mainEntryPointFilePath": "<projectFolder>/temp/types/api.d.ts",
    "dtsRollup": {
        "enabled": false
    },
    "messages": {
        "compilerMessageReporting": {
            "default": {
                "logLevel": "none"
            }
        }
    }
}
