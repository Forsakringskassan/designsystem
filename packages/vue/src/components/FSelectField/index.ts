export { default as FSelectField } from "./FSelectField.vue";
