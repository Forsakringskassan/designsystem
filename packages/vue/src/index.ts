export * from "./components";
export {
    config,
    type FKUIConfig,
    FKUIConfigButtonOrder,
    type FKUIContext,
    type MaybeWithFKUIContext,
    setRunningContext,
} from "./config";
export * from "./internal-components";
export * from "./plugins";
export * from "./types";
export * from "./utils";
