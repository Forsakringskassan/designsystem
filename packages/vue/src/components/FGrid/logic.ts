import { type Ref } from "vue";

export function isTableCell(
    el: Partial<HTMLTableCellElement> | null,
): el is HTMLTableCellElement & { parentElement: HTMLTableRowElement } {
    return el !== null && el.cellIndex !== undefined;
}

export function isAlphanumeric(keyCode: number): boolean {
    return (keyCode >= 48 && keyCode <= 57) || (keyCode >= 65 && keyCode <= 90);
}

export function getTableCell(
    e: KeyboardEvent,
): (HTMLTableCellElement & { parentElement: HTMLTableRowElement }) | undefined {
    const target = e.target as HTMLElement;
    return isTableCell(target)
        ? target
        : isTableCell(target?.parentElement)
          ? target.parentElement
          : undefined;
}

export function handleViewKeyDown(
    e: KeyboardEvent,
    activeCellIndex?: Ref<number>,
    activeRowIndex?: Ref<number>,
    lastCellIndex?: number,
    lastRowIndex?: number,
): void {
    if (
        activeCellIndex === undefined ||
        activeRowIndex === undefined ||
        lastCellIndex === undefined ||
        lastRowIndex === undefined
    ) {
        return;
    }

    const el = getTableCell(e);
    if (!el) {
        return;
    }

    if (e.code === "ArrowLeft") {
        e.preventDefault();
        if (el.cellIndex === 0) {
            return;
        }

        activeCellIndex.value = el.cellIndex - 1;
        activeRowIndex.value = el.parentElement.rowIndex;
    }

    if (e.code === "ArrowRight") {
        e.preventDefault();
        if (lastCellIndex === el.cellIndex) {
            return;
        }

        activeCellIndex.value = el.cellIndex + 1;
        activeRowIndex.value = el.parentElement.rowIndex;
    }

    if (e.code === "ArrowUp") {
        e.preventDefault();

        if (el.parentElement.rowIndex === 1) {
            return;
        }

        activeCellIndex.value = el.cellIndex;
        activeRowIndex.value = el.parentElement.rowIndex - 1;
    }

    if (e.code === "ArrowDown") {
        e.preventDefault();

        if (lastRowIndex === el.parentElement.rowIndex) {
            return;
        }

        activeCellIndex.value = el.cellIndex;
        activeRowIndex.value = el.parentElement.rowIndex + 1;
    }

    if (e.code === "Home") {
        e.preventDefault();

        if (e.ctrlKey) {
            activeCellIndex.value = 0;
            activeRowIndex.value = 1;
        } else {
            activeCellIndex.value = 0;
            activeRowIndex.value = el.parentElement.rowIndex;
        }
    }

    if (e.code === "End") {
        e.preventDefault();

        if (e.ctrlKey) {
            activeCellIndex.value = lastCellIndex;
            activeRowIndex.value = lastRowIndex;
        } else {
            activeCellIndex.value = lastCellIndex;
            activeRowIndex.value = el.parentElement.rowIndex;
        }
    }
}
