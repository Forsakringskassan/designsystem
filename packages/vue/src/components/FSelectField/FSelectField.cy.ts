import { defineComponent } from "vue";
import {
    DensityWrapper,
    densityWrapperHeight,
    densityWrapperWidth,
} from "@fkui/test-utils/vue";
import { ValidationService } from "@fkui/logic";
import { FSelectFieldPageObject } from "../../cypress";
import FTooltip from "../FTooltip/FTooltip.vue";
import FSelectField from "./FSelectField.vue";

describe("FSelectField", () => {
    it("should provide a page object that can access any necessary elements", () => {
        const TestComponent = defineComponent({
            template: /* HTML */ `
                <f-select-field
                    id="select-field"
                    v-model="model"
                    v-validation.required
                >
                    <template #label> Dropplista </template>
                    <template #tooltip>
                        <f-tooltip screen-reader-text="sr-text" header-tag="h1">
                            <template #header> Tooltip header </template>
                            <template #body> Tooltip body </template>
                        </f-tooltip>
                    </template>
                    <template #description="$scope">
                        <span :class="$scope.descriptionClass">
                            Hjälptext
                        </span>
                        <span :class="$scope.formatDescriptionClass">
                            Formatbeskrivning
                        </span>
                    </template>
                    <option disabled hidden value="">Gör ett val</option>
                    <option
                        v-for="option in options"
                        :key="option.value"
                        :value="option.value"
                    >
                        {{ option.label }}
                    </option>
                </f-select-field>
            `,
            components: {
                FSelectField,
                FTooltip,
            },
            data() {
                return {
                    model: "",
                    options: [
                        { value: "foo-value", label: "Foo label" },
                        { value: "bar-value", label: "Bar label" },
                    ],
                };
            },
            mounted() {
                ValidationService.setSubmitted("select-field");
            },
        });

        cy.mount(TestComponent);
        const selectField = new FSelectFieldPageObject(".select-field");
        selectField.label.el().should("contain.text", "Dropplista");
        selectField.label.description().should("contain.text", "Hjälptext");
        selectField.label
            .formatDescription()
            .should("contain.text", "Formatbeskrivning");
        selectField.arrowIcon().should("exist");

        selectField.dropdown().focus().blur();
        selectField.el().click();
        selectField.label
            .errorMessage()
            .should("exist")
            .should("contain.text", "Välj ett av alternativen.");

        selectField
            .dropdown()
            .should("have.value", null)
            .select("foo-value")
            .focus()
            .should("have.focus")
            .should("contain", "Foo label");

        selectField.label.errorMessage().should("not.exist");
        selectField.selectedOption().should("have.text", "Foo label");
        selectField.selectedValue().should("be.equal", "foo-value");
        selectField.numberOfOptions().should("be.equal", 2);

        selectField.tooltip.iButton().should("be.visible");
        selectField.tooltip.iButton().click();
    });

    /* eslint-disable-next-line mocha/no-skipped-tests -- temporary to get builds running */
    describe.skip("density", () => {
        const DensityComponent = defineComponent({
            template: /* HTML */ `
                <density-wrapper>
                    <f-select-field v-model="selectField">
                        <template #label> Dropplista </template>
                        <option value="Text">Text</option>
                        <option value="Text2">Text 2</option>
                        <option value="Text3">Text 3</option>
                    </f-select-field>
                </density-wrapper>
            `,
            components: {
                DensityWrapper,
                FSelectField,
            },
            data() {
                return { selectField: "Text" };
            },
        });

        it(`should be densified`, () => {
            cy.viewport(densityWrapperWidth, densityWrapperHeight);
            cy.mount(DensityComponent);
            cy.toMatchScreenshot();
        });
    });
});
