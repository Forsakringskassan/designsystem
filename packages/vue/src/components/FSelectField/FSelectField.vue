<script lang="ts">
import { defineComponent, inject } from "vue";
import { ElementIdService, type ValidityEvent } from "@fkui/logic";
import { FIcon } from "../FIcon";
import { FLabel } from "../FLabel";
import { renderSlotText, dispatchComponentValidityEvent } from "../../utils";
import { resolveWidthClass } from "./FSelectField.logic";

export default defineComponent({
    name: "FSelectField",
    components: { FIcon, FLabel },
    inheritAttrs: false,
    props: {
        /**
         * The id for the select id attribute.
         * The id for the label for attribute.
         * If the prop is not set a random value will be generated.
         */
        id: {
            type: String,
            required: false,
            default: () => ElementIdService.generateElementId(),
        },
        /**
         * Show the component inline.
         */
        inline: {
            type: Boolean,
            required: false,
            default: false,
        },
        /**
         * The value for the input.
         * If the prop is not set undefined will be used.
         * @model
         */
        modelValue: {
            type: [String, Number, Object, Array, Boolean],
            required: false,
            default: undefined,
        },
        /**
         * Set responsive width for label section.
         *
         * ```
         * label-width="md-9 lg-6"
         * ```
         */
        labelWidth: {
            type: String,
            required: false,
            default: "sm-12",
        },
        /**
         * Set responsive width for select section that wraps select element and icons.
         *
         * ```
         * select-width="md-6 lg-3"
         * ```
         */
        selectWidth: {
            type: String,
            required: false,
            default: "sm-12",
        },
    },
    emits: ["change", "update:modelValue"],
    setup() {
        return {
            textFieldTableMode: inject("textFieldTableMode", false) as boolean,
        };
    },
    data() {
        return {
            validityMode: "INITIAL" as string,
            validationMessage: "" as string,
        };
    },
    computed: {
        attrs(): Record<string, unknown> {
            return {
                ...this.$attrs,
                // Disable change
                onChange: () => undefined,
            };
        },
        hasError(): boolean {
            return this.validityMode === "ERROR";
        },
        rootClass(): Record<string, boolean> {
            return {
                ["select-field--error"]: this.hasError,
                ["select-field--inline"]: this.inline,
                ["text-field--table"]: this.textFieldTableMode,
                ["select-field--table-error"]: this.textFieldTableMode && this.hasError,
            };
        },
        labelClass(): string {
            return this.textFieldTableMode ? "sr-only" : "";
        },
        labelWrapperClass(): string | undefined {
            return resolveWidthClass(this.labelWidth, this.inline);
        },
        selectWrapperClass(): string | undefined {
            return resolveWidthClass(this.selectWidth, this.inline);
        },
        vModel: {
            get() {
                return this.modelValue;
            },
            set(value: unknown) {
                /**
                 * V-model event.
                 * @event update:modelValue
                 * @type {string}
                 */
                this.$emit("update:modelValue", value);

                /**
                 * Emitted when the value of the dropdown changes.
                 *
                 * @event change
                 * @type {string}
                 */
                this.$emit("change", value);
            },
        },
    },
    methods: {
        async onValidity({ detail }: CustomEvent<ValidityEvent>): Promise<void> {
            this.validationMessage = detail.validationMessage;
            this.validityMode = detail.validityMode;

            await this.$nextTick();

            const errorMessage = renderSlotText(this.$slots.label) ?? "";

            const element = this.$el.querySelector(`#${detail.elementId}`);
            if (element) {
                dispatchComponentValidityEvent(element, {
                    ...detail,
                    errorMessage,
                    focusElementId: detail.elementId,
                });
            }
        },
    },
});
</script>

<template>
    <div class="select-field" :class="rootClass" @validity="onValidity">
        <div :class="labelWrapperClass">
            <f-label :for="id" :class="labelClass">
                <template #default>
                    <!-- @slot Slot for label content. -->
                    <slot name="label"></slot>
                </template>

                <template v-if="$slots.tooltip" #tooltip>
                    <!-- @slot Slot for tooltip. -->
                    <slot name="tooltip"></slot>
                </template>

                <template #description="{ descriptionClass, formatDescriptionClass }">
                    <!--
                    @slot Optional slot for description. See {@link FLabel} for details.
                    @binding {string[]} description-class CSS classes for primary description content.
                    @binding {string[]} format-description-class CSS classes for format description.
                -->
                    <slot name="description" :description-class :format-description-class></slot>
                </template>

                <template #error-message>
                    <!--
                    @slot Slot for displaying single or several error messages.
                    @binding {boolean} hasError Set to true when a validation error is present
                    @binding {string} validationMessage Descriptive validation error message for current error
                -->
                    <slot name="error-message" v-bind="{ hasError, validationMessage }">
                        <template v-if="hasError">{{ validationMessage }}</template>
                    </slot>
                </template>
            </f-label>
        </div>
        <div class="select-field__icon-wrapper" :class="selectWrapperClass">
            <select :id="id" v-model="vModel" class="select-field__select" v-bind="attrs">
                <slot></slot>
            </select>
            <f-icon
                v-if="hasError && textFieldTableMode"
                ref="icon"
                class="text-field__icon input-icon select-field__error-popup-icon"
                name="error"
            ></f-icon>
            <f-icon class="select-field__icon" name="arrow-down"></f-icon>
        </div>
    </div>
</template>
