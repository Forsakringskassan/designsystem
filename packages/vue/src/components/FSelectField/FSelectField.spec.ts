import "html-validate/jest";
import { defineComponent, h } from "vue";
import { VueWrapper, mount } from "@vue/test-utils";
import flushPromises from "flush-promises";
import { type ValidityEvent, type ValidatableHTMLElement } from "@fkui/logic";
import FSelectField from "./FSelectField.vue";

function createTestComponentWithOptions(
    options: Array<{ text: string; value: unknown }> = [],
): ReturnType<typeof defineComponent> {
    return defineComponent({
        components: { FSelectField },
        render() {
            const children = options.map((option) => {
                const data = { value: option.value };
                return h("option", data, option.text);
            });
            return h(FSelectField, this.$attrs, () => children);
        },
    });
}

function createWrapper({
    props = {},
    slots = {},
    attrs = {},
} = {}): VueWrapper {
    return mount(FSelectField, {
        attrs: { ...attrs, id: "select-field" },
        props: { ...props },
        slots: {
            default: /* HTML */ `
                <option>Apple</option>
                <option>Banana</option>
            `,
            label: "Fruit",
            ...slots,
        },
        global: {
            stubs: ["f-icon"],
        },
    });
}

describe("snapshots", () => {
    it("should match snapshot with label and select", () => {
        const wrapper = createWrapper();

        expect(wrapper.element).toMatchSnapshot();
    });

    it("should match snapshot with label, select and error message", () => {
        const wrapper = createWrapper({
            attrs: { "aria-invalid": true },
            slots: { "error-message": "ERROR_MESSAGE" },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it("should match snapshot with label, tooltip, description, error message and select", () => {
        const wrapper = createWrapper({
            slots: {
                description: "DESCRIPTION",
                tooltip: "TOOLTIP",
                "error-message": "ERROR_MESSAGE",
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it.each`
        validityMode | isValid
        ${"VALID"}   | ${true}
        ${"ERROR"}   | ${false}
        ${"INTIAL"}  | ${true}
        ${"INITIAL"} | ${false}
    `(
        "should match snapshot when validityMode is $validityMode and isValid is $isValid",
        async ({ validityMode, isValid }) => {
            const wrapper = createWrapper({
                attrs: { id: "elementId" },
            });

            wrapper.element.dispatchEvent(
                new CustomEvent<ValidityEvent>("validity", {
                    detail: {
                        target: wrapper.element as ValidatableHTMLElement,
                        elementId: "elementId",
                        isValid,
                        validityMode,
                        validationMessage: "Something went wrong.",
                        nativeEvent: "change",
                    },
                }),
            );
            await flushPromises();
            wrapper.vm.$forceUpdate();

            expect(wrapper.element).toMatchSnapshot();
        },
    );
});

describe("attributes", () => {
    it("should pass attributes", () => {
        const wrapper = createWrapper({
            attrs: {
                disabled: true,
                required: true,
            },
        });
        const select = wrapper.get("select");

        expect(select.attributes("disabled")).toBeDefined();
        expect(select.attributes("required")).toBeDefined();
    });
});

describe("inline", () => {
    it("should not set class by default", () => {
        const wrapper = createWrapper();
        expect(wrapper.classes()).not.toContain("select-field--inline");
    });

    it("should set class when enabled", () => {
        const wrapper = createWrapper({
            props: {
                inline: true,
            },
        });

        expect(wrapper.classes()).toContain("select-field--inline");
    });
});

describe("events", () => {
    it("should pass listeners", async () => {
        const foobar = jest.fn();
        const wrapper = createWrapper({
            attrs: { onFoobar: foobar },
        });
        const element = wrapper.get("select");
        await element.trigger("foobar");
        expect(foobar).toHaveBeenCalled();
    });

    it("should support v-model by emitting update:modelValue event with string", () => {
        const wrapper = mount(
            createTestComponentWithOptions([
                { text: "Banana", value: "banana" },
                { text: "Apple", value: "apple" },
            ]),
            { props: { modelValue: "banana" } },
        );
        const select = wrapper.get("select");
        const htmlSelect = select.element;

        expect(htmlSelect.value).toBe("banana");
        select.setValue("apple");
        expect(htmlSelect.value).toBe("apple");
        expect(
            wrapper
                .findComponent(FSelectField)
                .emitted("update:modelValue")![0][0],
        ).toMatchInlineSnapshot(`"apple"`);
    });

    it("should support v-model by emitting update:modelValue event with object", async () => {
        const wrapper = mount(
            createTestComponentWithOptions([
                { text: "BananaObject", value: { id: 1, fruit: "banana" } },
                { text: "AppleObject", value: { id: 2, fruit: "apple" } },
            ]),
            { props: { modelValue: { id: 1, fruit: "banana" } } },
        );
        await wrapper.vm.$nextTick();
        const vModelValue = wrapper
            .findComponent(FSelectField)
            .props("modelValue");

        expect(vModelValue).toEqual({ id: 1, fruit: "banana" });
    });

    it("should emit change event with when value changes", () => {
        const wrapper = mount(
            createTestComponentWithOptions([
                { text: "Banana", value: "banana" },
                { text: "Apple", value: "apple" },
            ]),
            { props: { modelValue: "banana" } },
        );
        const select = wrapper.get("select");
        select.setValue("apple");
        expect(
            wrapper.findComponent(FSelectField).emitted("change")![0][0],
        ).toMatchInlineSnapshot(`"apple"`);
    });
});

describe("html-validate", () => {
    it.each`
        html
        ${"<f-select-field><template #label>label</template><option>Apple</option></f-select-field>"}
        ${"<f-select-field><template #label>label</template><template #default>default</template><option>Apple</option></f-select-field>"}
        ${'<f-select-field name="select-field"><template #label>label</template><option>Apple</option></f-select-field>'}
    `("$html should be valid", ({ html }) => {
        expect.assertions(1);

        expect(html).toHTMLValidate();
    });

    it.each`
        html
        ${"<f-select-field><template #tooltip><div /></template><template #label>label</template></f-select-field>"}
        ${"<f-select-field><template #label>label</template><template #default><div></div></template><option>Apple</option></f-select-field>"}
        ${"<f-select-field><template #label>label</template><div></div></f-select-field>"}
    `("$html should be invalid", ({ html }) => {
        expect.assertions(3);
        let catchedError;

        try {
            expect(html).toHTMLValidate();
        } catch (error) {
            catchedError = error;
        } finally {
            expect(catchedError).toBeDefined();
            expect(catchedError).toMatchSnapshot();
        }
    });
});
