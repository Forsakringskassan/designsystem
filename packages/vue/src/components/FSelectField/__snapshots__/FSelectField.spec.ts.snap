// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html-validate <f-select-field><template #label>label</template><div></div></f-select-field> should be invalid 1`] = `
[Error: Expected HTML to be valid but had the following errors:

  <div> element is not permitted as content under <f-select-field> [element-permitted-content]]
`;

exports[`html-validate <f-select-field><template #label>label</template><template #default><div></div></template><option>Apple</option></f-select-field> should be invalid 1`] = `
[Error: Expected HTML to be valid but had the following errors:

  <div> element is not permitted as content under slot "default" (<f-select-field>) [element-permitted-content]]
`;

exports[`html-validate <f-select-field><template #tooltip><div /></template><template #label>label</template></f-select-field> should be invalid 1`] = `
[Error: Expected HTML to be valid but had the following errors:

  <div> element is not permitted as content under slot "tooltip" (<f-select-field>) [element-permitted-content]]
`;

exports[`snapshots should match snapshot when validityMode is ERROR and isValid is false 1`] = `
<div
  class="select-field select-field--error"
>
  <div
    class="i-width-sm-12"
  >
    <label
      class="label"
      for="select-field"
    >
      
      
      Fruit
      
      
      
      
      <span
        class="label__message label__message--error"
      >
        <f-icon-stub
          class="label__icon--left"
          library="f"
          name="error"
        />
        
        
        
        Something went wrong.
        
        
        
      </span>
    </label>
  </div>
  <div
    class="select-field__icon-wrapper i-width-sm-12"
  >
    <select
      class="select-field__select"
      id="select-field"
    >
      
      <option>
        Apple
      </option>
      <option>
        Banana
      </option>
      
    </select>
    <!--v-if-->
    <f-icon-stub
      class="select-field__icon"
      library="f"
      name="arrow-down"
    />
  </div>
</div>
`;

exports[`snapshots should match snapshot when validityMode is INITIAL and isValid is false 1`] = `
<div
  class="select-field"
>
  <div
    class="i-width-sm-12"
  >
    <label
      class="label"
      for="select-field"
    >
      
      
      Fruit
      
      
      
      
      <!--v-if-->
    </label>
  </div>
  <div
    class="select-field__icon-wrapper i-width-sm-12"
  >
    <select
      class="select-field__select"
      id="select-field"
    >
      
      <option>
        Apple
      </option>
      <option>
        Banana
      </option>
      
    </select>
    <!--v-if-->
    <f-icon-stub
      class="select-field__icon"
      library="f"
      name="arrow-down"
    />
  </div>
</div>
`;

exports[`snapshots should match snapshot when validityMode is INTIAL and isValid is true 1`] = `
<div
  class="select-field"
>
  <div
    class="i-width-sm-12"
  >
    <label
      class="label"
      for="select-field"
    >
      
      
      Fruit
      
      
      
      
      <!--v-if-->
    </label>
  </div>
  <div
    class="select-field__icon-wrapper i-width-sm-12"
  >
    <select
      class="select-field__select"
      id="select-field"
    >
      
      <option>
        Apple
      </option>
      <option>
        Banana
      </option>
      
    </select>
    <!--v-if-->
    <f-icon-stub
      class="select-field__icon"
      library="f"
      name="arrow-down"
    />
  </div>
</div>
`;

exports[`snapshots should match snapshot when validityMode is VALID and isValid is true 1`] = `
<div
  class="select-field"
>
  <div
    class="i-width-sm-12"
  >
    <label
      class="label"
      for="select-field"
    >
      
      
      Fruit
      
      
      
      
      <!--v-if-->
    </label>
  </div>
  <div
    class="select-field__icon-wrapper i-width-sm-12"
  >
    <select
      class="select-field__select"
      id="select-field"
    >
      
      <option>
        Apple
      </option>
      <option>
        Banana
      </option>
      
    </select>
    <!--v-if-->
    <f-icon-stub
      class="select-field__icon"
      library="f"
      name="arrow-down"
    />
  </div>
</div>
`;

exports[`snapshots should match snapshot with label and select 1`] = `
<div
  class="select-field"
>
  <div
    class="i-width-sm-12"
  >
    <label
      class="label"
      for="select-field"
    >
      
      
      Fruit
      
      
      
      
      <!--v-if-->
    </label>
  </div>
  <div
    class="select-field__icon-wrapper i-width-sm-12"
  >
    <select
      class="select-field__select"
      id="select-field"
    >
      
      <option>
        Apple
      </option>
      <option>
        Banana
      </option>
      
    </select>
    <!--v-if-->
    <f-icon-stub
      class="select-field__icon"
      library="f"
      name="arrow-down"
    />
  </div>
</div>
`;

exports[`snapshots should match snapshot with label, select and error message 1`] = `
<div
  class="select-field"
>
  <div
    class="i-width-sm-12"
  >
    <label
      class="label"
      for="select-field"
    >
      
      
      Fruit
      
      
      
      
      <span
        class="label__message label__message--error"
      >
        <f-icon-stub
          class="label__icon--left"
          library="f"
          name="error"
        />
        
        
        ERROR_MESSAGE
        
        
      </span>
    </label>
  </div>
  <div
    class="select-field__icon-wrapper i-width-sm-12"
  >
    <select
      aria-invalid="true"
      class="select-field__select"
      id="select-field"
    >
      
      <option>
        Apple
      </option>
      <option>
        Banana
      </option>
      
    </select>
    <!--v-if-->
    <f-icon-stub
      class="select-field__icon"
      library="f"
      name="arrow-down"
    />
  </div>
</div>
`;

exports[`snapshots should match snapshot with label, tooltip, description, error message and select 1`] = `
<div
  class="select-field"
>
  <div
    class="i-width-sm-12"
  >
    <div
      class=""
    >
      <div
        class="tooltip-before"
      >
        <label
          class="label tooltip-before__label"
          for="select-field"
        >
          
          
          Fruit
          
          
        </label>
      </div>
      
      
      TOOLTIP
      
      
      <label
        class="label sr-separator"
        for="select-field"
      >
        
        
        DESCRIPTION
        
        
        <span
          class="label__message label__message--error"
        >
          <f-icon-stub
            class="label__icon--left"
            library="f"
            name="error"
          />
          
          
          ERROR_MESSAGE
          
          
        </span>
      </label>
    </div>
  </div>
  <div
    class="select-field__icon-wrapper i-width-sm-12"
  >
    <select
      class="select-field__select"
      id="select-field"
    >
      
      <option>
        Apple
      </option>
      <option>
        Banana
      </option>
      
    </select>
    <!--v-if-->
    <f-icon-stub
      class="select-field__icon"
      library="f"
      name="arrow-down"
    />
  </div>
</div>
`;
